### ISSUE TITLE
Bug: sam list stack-outputs - KeyError

### ISSUE THREAD

**Issue by Tak1wa** [aws/aws-sam-cli#4658]
### Description:

sam list is a nice feature.
I was trying sam list stack-outputs on various stacks.
I noticed an error in some of the CloudFormation stacks.

### Steps to reproduce:

- Follow the steps below to deploy the base stack to use the AWS Toolkit for .NET Refactoring.
  - https://docs.aws.amazon.com/tk-dotnet-refactoring/latest/userguide/roles-and-policies.html
- Run `sam list stack-outputs --stack-name refactoringtoolkit`

### Observed result:

```
iwasa.takahito@xxx hoge0204sam-dotnet-serverless % sam list stack-outputs --stack-name refactoringtoolkit --d
ebug
2023-02-04 06:55:14,894 | Config file location: /Users/iwasa.takahito/work/hoge0204sam/hoge0204sam-dotnet-serverless/samconfig.toml
2023-02-04 06:55:14,894 | Config file '/Users/iwasa.takahito/work/hoge0204sam/hoge0204sam-dotnet-serverless/samconfig.toml' does not exist
2023-02-04 06:55:14,922 | Using config file: samconfig.toml, config environment: default
2023-02-04 06:55:14,922 | Expand command line arguments to:
2023-02-04 06:55:14,922 | --stack_name=refactoringtoolkit --output=table 
2023-02-04 06:55:15,268 | Telemetry endpoint configured to be https://aws-serverless-tools-telemetry.us-west-2.amazonaws.com/metrics
2023-02-04 06:55:15,268 | Template is not provided in context, skip adding project type metric
2023-02-04 06:55:15,321 | Sending Telemetry: {'metrics': [{'commandRun': {'requestId': '5a7c06ce-77fa-43f9-acd4-3614c6e187f6', 'installationId': '0f5b04cf-df1a-4263-aef3-5eee2c8c8c43', 'sessionId': '3cc4e82e-c793-47d0-a926-b07cc089c954', 'executionEnvironment': 'CLI', 'ci': False, 'pyversion': '3.8.16', 'samcliVersion': '1.72.0', 'awsProfileProvided': False, 'debugFlagProvided': True, 'region': '', 'commandName': 'sam list stack-outputs', 'metricSpecificAttributes': {'gitOrigin': None, 'projectName': '5bdffcbe2506879e44ea1e6791554ad1fbd86d4577bff7764b38b2de6c3275bb', 'initialCommit': None}, 'duration': 346, 'exitReason': 'KeyError', 'exitCode': 255}}]}
2023-02-04 06:55:15,989 | HTTPSConnectionPool(host='aws-serverless-tools-telemetry.us-west-2.amazonaws.com', port=443): Read timed out. (read timeout=0.1)

Error: 'Description'
Traceback:
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/click/decorators.py", line 84, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/samcli/lib/telemetry/metric.py", line 183, in wrapped
    raise exception  # pylint: disable=raising-bad-type
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/samcli/lib/telemetry/metric.py", line 150, in wrapped
    return_value = func(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/samcli/lib/utils/version_checker.py", line 41, in wrapped
    actual_result = func(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/samcli/cli/main.py", line 92, in wrapper
    return func(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/samcli/commands/_utils/command_exception_handler.py", line 40, in wrapper_command_exception_handler
    raise ex
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/samcli/commands/_utils/command_exception_handler.py", line 25, in wrapper_command_exception_handler
    return func(*args, **kwargs)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/samcli/commands/list/stack_outputs/command.py", line 40, in cli
    do_cli(stack_name=stack_name, output=output, region=self.region, profile=self.profile)
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/samcli/commands/list/stack_outputs/command.py", line 52, in do_cli
    stack_output_context.run()
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/samcli/commands/list/stack_outputs/stack_outputs_context.py", line 45, in run
    producer.produce()
  File "/opt/homebrew/Cellar/aws-sam-cli/1.72.0/libexec/lib/python3.8/site-packages/samcli/lib/list/stack_outputs/stack_outputs_producer.py", line 66, in produce
    Description=stack_output["Description"],

An unexpected error was encountered while executing "sam list stack-outputs".
Search for an existing issue:
https://github.com/aws/aws-sam-cli/issues?q=is%3Aissue+is%3Aopen+Bug%3A%20sam%20list%20stack-outputs%20-%20KeyError
Or create a bug report:
https://github.com/aws/aws-sam-cli/issues/new?template=Bug_report.md&title=Bug%3A%20sam%20list%20stack-outputs%20-%20KeyError
```

### Expected result:

I hope to get the output of some CloudFormation stacks

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: macOS Ventura 13.1, Apple M1 Max
2. `sam --version`:SAM CLI, version 1.72.0
3. AWS region:ap-northeast-1

```
{
  "version": "1.72.0",
  "system": {
    "python": "3.8.16",
    "os": "macOS-13.1-arm64-arm-64bit"
  },
  "additional_dependencies": {
    "docker_engine": "20.10.17",
    "aws_cdk": "2.50.0 (build 4c11af6)",
    "terraform": "1.3.4"
  }
}
```

**Comment by mildaniel** [comment]
Thanks @Tak1wa for bringing this to our attention. I'm going to look into what's happening here.

**Comment by mildaniel** [comment]
Raised a pull request to address this issue.

**Comment by mndeveci** [comment]
Released with [v1.73.0](https://github.com/aws/aws-sam-cli/releases/tag/v1.73.0)
